src/lib/keccak.ts

```diff
@@ -209,18 +389,233 @@ function permutation(state: Field[][], rc: Field[]): Field[][] {
   );
 }
 
-// TESTING
+// KECCAK SPONGE
 
-const blockTransformation = (state: Field[][]): Field[][] =>
-  permutation(state, ROUND_CONSTANTS);
+// Absorb padded message into a keccak state with given rate and capacity
```

Comment:
Again, I would prefer to only allow multiples of 8 here. So either assert it, or just use bytes by default. This is because we don't give support for non-byte inputs nor parameters.